nrf_wifi/hw_if/hal/src/hal_api.c

```diff
@@ -1044,6 +1044,39 @@ enum nrf_wifi_status hal_rpu_eventq_process(struct nrf_wifi_hal_dev_ctx *hal_dev
 	return status;
 }
 
+enum nrf_wifi_status hal_rpu_eventq_drain(struct nrf_wifi_hal_dev_ctx *hal_dev_ctx)
+{
+	enum nrf_wifi_status status = NRF_WIFI_STATUS_SUCCESS;
+	struct nrf_wifi_hal_msg *event = NULL;
+	unsigned long flags = 0;
+
+	while (1) {
+		nrf_wifi_osal_spinlock_irq_take(hal_dev_ctx->hpriv->opriv,
+						hal_dev_ctx->lock_rx,
+						&flags);
+
+		event = nrf_wifi_utils_q_dequeue(hal_dev_ctx->hpriv->opriv,
+						 hal_dev_ctx->event_q);
+
+		nrf_wifi_osal_spinlock_irq_rel(hal_dev_ctx->hpriv->opriv,
+					       hal_dev_ctx->lock_rx,
+					       &flags);
+
+		if (!event) {
+			goto out;
```

Comment:
Shouldn't status be updated to `FAIL` here?

Reply:
No, not having any more events is a positive case.